tests/ui/const-generics/generic_arg_infer/infer-arg-test.stderr

```diff
@@ -10,14 +10,14 @@ error: expected identifier, found reserved identifier `_`
 LL | fn bad_infer_fn<_>() {}
    |                 ^ expected identifier, found reserved identifier
 
-error[E0392]: parameter `_` is never used
+error[E0392]: type parameter `_` is never used
```

Comment:
I could extra code for skipping this error entirely by delaying a bug on `_` when checking the “usedness”. Not sure if it's worth it though.

Reply:
It may be very hard to track the `expected identifier, found reserved identifier` error all the way into the type system. We'd probably have to add a `GenericParamDefKind::Error` to do that. May be worth exploring, but not in this PR